src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,295 @@
+            end
+            continue
+        end
+        # Find the index in the intr_list (same as b_idx_list) where the inter point is
+        # TODO:find all is inefficient though, so it might be better to make dictionary of indices
+        i = findall(x->x==b_list[ii].idx, b_idx_list)
+        if !isempty(i)     
+            # Order intersection points based on alpha values
+            new_order = sortperm(alpha_b_list[i])
+            pts_to_add = Array{PolyNode, 1}(undef, length(i))
+            for m in eachindex(i)
+                pts_to_add[new_order[m]] = PolyNode(i[m], true, a_idx_list[i[m]], false, alpha_b_list[i[m]])
+                b_neighbors[i[m]] = ii + new_order[m]
+            end   
+            # I use splice instead of insert so I can insert array   
+            splice!(b_list, ii+1:ii, pts_to_add)
+            skip = true
+            num_skips = length(i)
+        end
+    end
+
+    b_neighbors = b_neighbors[1:counter-1]
+
+    return b_neighbors, b_list
+end
+
+"""
+    _flag_ent_exit(
+        ::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, 
+        poly_b, a_list, b_list, edges_a, edges_b
+    )::a_list, b_list
+
+    This function flags all the intersection points as either an 'entry' or 'exit' point.
+"""
+function _flag_ent_exit(::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b, a_list, b_list, edges_a, edges_b)
+    # Put in ent exit flags for poly_a
+    status = false
+    for ii in eachindex(a_list)
+        if ii == 1
+            temp = point_in_polygon(edges_a[ii][1], poly_b)
+            status = !(temp[1])
+            continue
+        end
+        if a_list[ii].inter
+            a_list[ii].ent_exit = status
+            status = !status
+        end
+    end
+
+    # Put in ent exit flags for poly_b
+    status = false
+    for ii in eachindex(b_list)
+        if ii == 1
+            temp = point_in_polygon(edges_b[ii][1], poly_a)
+            status = !(temp[1])
+            continue
+        end
+        if b_list[ii].inter
+            b_list[ii].ent_exit = status
+            status = !status
+        end
+    end
+
+    return a_list, b_list
+end
+
+"""
+    _build_ab_list(
+        ::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b
+    )::a_list, b_list, a_idx_list, intr_list, edges_a, edges_b
+
+    This function calls '_build_a_list', '_build_b_list', and '_flag_ent_exit'
+    in order to fully form a_list and b_list.
+"""
+function _build_ab_list(::GI.PolygonTrait, poly_a, ::GI.PolygonTrait, poly_b)
+    # Make a list for nodes of each polygon. Note the definition of PolyNode
+    a_list = Array{PolyNode, 1}(undef, _nedge(poly_a))
+    b_list = Array{PolyNode, 1}(undef, _nedge(poly_b))
+
+    # Initialize arrays to keep track of the important information 
+    # associated with the intersection points of poly_a and poly_b.
+    # I initialize these arrays assumed there will be a maximum of 
+    # 30 intersection points and then I truncate the arrays later.
+    k = 30
```

Comment:
Lets assume wayyyyy less points for now. This would have to be a really, really wild shape. Let's maybe make it a user parameter (n_inters = 2) for now and let's say like... 2? Or 4 if we want to be a little more generous.